### ISSUE TITLE
ingestIfNotExists doesn't work as expected

### ISSUE THREAD

**Issue by Ceridan** [Azure/azure-kusto-spark#157]
**Describe the bug**
I am trying to ingest data from Delta table format to Kusto (streaming). I am using `ingestByTags` property to avoid duplication. If I re-run pipleline on the exactly same data second time, I will get duplicates in the target Kusto table despite on I am using `ingestIfNotExists` property.

**To Reproduce**
Here is my Python code example, but tested both with Python and Scala.

```python
import json


source_df = (
  spark
  .readStream
  .format("delta").load(DATA_PATH)
  .coalesce(1)
)

kusto_options = {
  "cluster": KUSTO_CLUSTER,
  "database" : KUSTO_DATABASE,
  "table" : KUSTO_TABLE,
  "app_id": KUSTO_APP_ID,
  "app_secret": KUSTO_APP_SECRET,
  "authority_id": KUSTO_AUTHORITY_ID,
}

ingestion_props = {
  "ingestByTags": ["myTag"],
  "ingestIfNotExists": ["myTag"],
}

kusto_q = (
  source_df
  .writeStream
  .format("com.microsoft.kusto.spark.datasink.KustoSinkProvider")
  .option("checkpointLocation", KUSTO_CHECKPOINTS_PATH)
  .option("kustoCluster", kusto_options["cluster"])
  .option("kustoDatabase", kusto_options["database"])
  .option("kustoTable", kusto_options["table"])
  .option("kustoAadAppId", kusto_options["app_id"])
  .option("kustoAadAppSecret", kusto_options["app_secret"])
  .option("kustoAadAuthorityID", kusto_options["authority_id"])
  .option("writeEnableAsync", False)
  .option("sparkIngestionPropertiesJson", json.dumps(ingestion_props))  
  .trigger(once=True)
)

kusto_q.start().awaitTermination()
```

We consider here that source data doesn't change at all. If I run exactly the same code twice, I will get duplicates. One important thing here: uploaded extents has proper tags:  `ingest-by:myTag`.

**Expected behavior**
I expected that my data will be uploaded only once. The second attempt should be rejected because extent with tag `myTag` already exists.

**Screenshots**
Here is result of two sequential attempts to upload data:
![kustosink](https://user-images.githubusercontent.com/408149/97872398-033a1400-1d27-11eb-920b-73d410b3152a.png)

**Desktop (please complete the following information):**
- OS: `Linux 1029-141313-brunt80-10-200-2-5 4.15.0-1092-azure #102~16.04.1-Ubuntu SMP Tue Jul 14 20:28:23 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
- Version: `Apache Spark 2.4.5, Scala 2.11, Python 3.7.3`
- Package: `com.microsoft.azure.kusto:kusto-spark_2.4_2.11:2.3.0`
- Platform: `Azure Databricks`

**Comment by ohadbitt** [comment]
Hello,
This is a known issue where in certain scenarios the ingest-by tags were dropped, and was fixed in the last version (2.3.1). Please update or if you can't please try with adding 'flushImmediately'=true to the sparkIngestionProperties

**Comment by Ceridan** [comment]
We have checked this code behavior with latest 2.3.2 version. Now it works as expected, the problem is fixed in 2.3.2. Thank you!

**Comment by Ceridan** [comment]
Closed by #159
